src/pages/billing/Project.js

```diff
@@ -361,7 +363,7 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
             }), [7, 30, 90]),
             onChange: ({ value: selectedDays }) => {
               setSpendReportLengthInDays(selectedDays)
-              setTotalCost(null) // This will force the report to be recalculated based on selectedDays
```

Comment:
Changing the total cost itself does not cause the report to be recalculated-- `spendReportLengthInDays` does through the effect dependency.